### PR TITLE
Eliminate test configuration model 

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

_Select the type of your PR_

- Refactoring

### Description

The code base had two parallel configuration models, one in the project proper and an second one used by the tests.  These were kept in sync by hand. This duplication represented a maintenance overhead.

This change:

* Eliminates the test-only model.
* Fluent API allow test authors to create the configuration is retained (albeit with a few breaking changes _affecting the kroxylicious test author persona only_).  These continues to be auto-generated using Sundrio Builders.
* Fluent API continues to be a test-only API
* Hand rolled builders are provided for the pluggable - service loader - parts of the model (filters, micrometer, cluster network address providers). Unfortunately I could not find a reasonable way to get these builders to play really nicely with Sundrio generated builders.

Fundamental to the approach behind this change was the ability to serialise/deserialise the config model with fidelity. To achieve this it was necessary to tease apart the concept of the config model from the runtime entity model.  In concrete terms, this means `VirtualCluster` config objects are used to create `VirtualCluster` model objects in the same way `FilterDefinitions` are materialised into `Filters`.

### Additional Context

_Why are you making this pull request?_

### CODE REVIEW

**Comment by robobario** [comment]
So if I read it right the builders and sundr dependencies still live in kroxylicious-test-tools, would it make sense to move them into `kroxylicious`? I'm thinking about if someone wanted to programmatically embed Kroxylicious, it would be nice to have access to fluent builders without needing to depend on test tools.

**Comment by k-wall** [comment]
> So if I read it right the builders and sundr dependencies still live in kroxylicious-test-tools

Right. With this PR, it remains a test only API. I didn't want to take the step to making it public API without a conversation amongst the team.  

>, would it make sense to move them into `kroxylicious`? 

Yes, we could do that.  The `ExternalBuildables` annotations just need to change into `Buildable` annotations and be placed on the resources themselves.  I actually tested this out already and verified this would work.

> I'm thinking about if someone wanted to programmatically embed Kroxylicious, it would be nice to have access to fluent builders without needing to depend on test tools.

I had thought about this use case too.

What's making me hesitate about making it public API is the fact that the hand rolled builders don't play seamlessly with the auto-generated ones.

we currently have:

```
.addToFilters(new FilterDefinitionBuilder("foo").withConfig("a","b"build())
```

what I really want:

```
.addNewFilter()
.withType("MyFilter")
.withConfig("a", "b")
.endFilter();  // the builder would have custom smarts to invoke our service loader and reject the case where type had not been provided.
```

Unfortunately, as far as I can see, sundio doesn't let you provide your own builder or substitute your own build method.

**Review by SamBarker — APPROVED** [review]
I'm with @k-wall that the builder API could be better but otherwise I'm +1.

I'm slightly surprised and disappointed that this diff is only a net -45 lines, but such is life.

**Review by robobario — APPROVED** [review]
LGTM, have confirmed we can depend on kroxylicious-test-tools externally without a dependency on sundr-core resolving #371 :+1:

File: kroxylicious/src/main/java/io/kroxylicious/proxy/config/ConfigParser.java [hunk 1]
```diff
@@ -53,4 +46,27 @@ public Configuration parseConfiguration(InputStream configuration) {
...
+    public String toYaml(Configuration configuration) {
+        try {
+            return mapper.writeValueAsString(configuration);
+        }
+        catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static ObjectMapper createObjectMapper() {
```
[the remark this answers is not in the document]
Comment (k-wall) [reply kroxylicious/src/main/java/io/kroxylicious/proxy/config/ConfigParser.java:59]: ObjectMappers are mutable (registerModule, setTimeZone etc), so it is probably safest if continue to hand the calls their own instance. I will make ConfigParser's own mapper a static final.

[the remark this answers is not in the document]
Comment (SamBarker) [reply kroxylicious/src/main/java/io/kroxylicious/proxy/config/ConfigParser.java:59]: But don't we want a single consistent set of objection mapper capabilities to be used? 

The fact it's mutable is annoying for that purpose. 

Obviously if we were handing them to filters then they would need to be scoped to each filter/filter contributor

**Pull request merged**